pkg/v1/remote/index.go

```diff
@@ -182,3 +182,48 @@ func (r *remoteIndex) childDescriptor(child v1.Descriptor, platform v1.Platform)
 		platform:   platform,
 	}, nil
 }
+
+// Check if the given platform can run the required platform.
+// Matching platforms must have identical architecture, OS,
+// OS version (if provided), and variant (if provided)
+// Required features must be a subset of the features in the given platform.
+func (r *remoteIndex) matchesPlatform(given v1.Platform, required v1.Platform) bool {
+	// required fields
+	if given.Architecture != required.Architecture || given.OS != required.OS {
+		return false
+	}
+
+	// optional fields
+	if given.OSVersion != "" && given.OSVersion != required.OSVersion {
+		return false
+	}
+	if given.Variant != "" && given.Variant != required.Variant {
```

Comment:
If I understand correctly, `required` is the one that user inputs? and `given` is from the manifest list, then we should check if `required.Variant` (and `required.OSVersion`, etc) is not empty instead of the given one?